### ISSUE TITLE
Error: closeTransport called before connection was established

### ISSUE THREAD

**Issue by ChenHaoJie9527** [WalletConnect/walletconnect-monorepo#2392]
### Link to minimal reproducible example

Initialize web3Modal with an error!

### Summary

![image](https://github.com/WalletConnect/web3modal/assets/68054968/c04f0771-50fb-41b6-af2e-5eb794edfad8)

### List of related npm package versions

import {
  EthereumClient,
  w3mConnectors,
  w3mProvider,
} from "@web3modal/ethereum";
import { Web3Modal } from "@web3modal/html";
import { configureChains, createConfig, mainnet } from "@wagmi/core";
// import { arbitrum, mainnet, polygon } from "@wagmi/core/chains";

// const chains = [arbitrum, mainnet, polygon];
const projectId = "349dd70c82d590a2a01a5e31da8ca2b2";

const { publicClient, chains, webSocketPublicClient } = configureChains(
  [mainnet],
  [w3mProvider({ projectId })]
);
const wagmiConfig = createConfig({
  autoConnect: true,
  connectors: w3mConnectors({ projectId, version: 2, chains }),
  publicClient,
  webSocketPublicClient,
});
const ethereumClient = new EthereumClient(wagmiConfig, chains);
export const web3modal = new Web3Modal({ projectId }, ethereumClient);

**Comment by xzilja** [comment]
Will transfer this to our other repository, where it is being addressed, ty for report

**Comment by ganchoradkov** [comment]
Hey @ChenHaoJie9527, are you using some kind of VPN? 
Which region are you trying to connect from?

**Comment by qqqzhch** [comment]
I also encountered this problem
```
const initializeWalletConnectClient = async () => {
      const core = new Core({
        projectId: WALLETCONNECT_V2_PROJECT_ID
      })

      const web3wallet = await Web3Wallet.init({
        core,
        metadata: Vault_METADATA
      })

      setWeb3wallet(web3wallet)
    }
```
Turning vpn on or off will cause this problem
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/3917994/7bc21be4-efdb-44db-88f0-3b2fe3e33463)

**Comment by ganchoradkov** [comment]
>Turning vpn on or off will cause this problem

Does the connection recover by itself as it should?
Turning on/off the vpn might kill the socket connection silently without emitting disconnect events thus leading to the reconnection issue

**Comment by ylmin** [comment]
I also encountered this problem

**Comment by AdamK222** [comment]
I'm also seeing this

**Comment by qqqzhch** [comment]
I changed to a new vpn and the error disappeared

**Comment by amerqureshi** [comment]
also seeing this error

**Comment by script-money** [comment]
I have same problem, it solve by open system proxy

**Comment by bkrem** [comment]
Hi everyone,

The error messaging here is a bit confusing (we're looking into making this clearer), but the key error log here is actually `Error: socket stalled`, which precedes the `closeTransport` one. 

This simply means that for some reason the WalletConnect client wasn't able to (re-)establish a connection to the relay. 

Reasons for this could be varied:
- issue with your local connection to the relay, e.g. due to geographical factors, DNS, etc...
- a reconnection edge case, as some here noted regarding switching between VPN connections
- possibly some edge case in relay availability/behaviour

If someone can provide a consistent reproduction, the team can dig into this more and try to squash out any edge cases we can control 🙏

**Comment by scholtz** [comment]
I am also seeing this.. 

```
closeTransport called before connection was established
    at EventEmitter.rejectTransportOpen (webpack-internal:///./node_modules/@walletconnect/core/dist/index.es.js:81:39042)
    at Object.onceWrapper (webpack-internal:///./node_modules/events/events.js:242:28)
    at EventEmitter.emit (webpack-internal:///./node_modules/events/events.js:153:5)
    at Pt.transportOpen (webpack-internal:///./node_modules/@walletconnect/core/dist/index.es.js:81:38593)
    at async Promise.all (index 1)
    at async Pt.init (webpack-internal:///./node_modules/@walletconnect/core/dist/index.es.js:81:36524)
    at async J.initialize (webpack-internal:///./node_modules/@walletconnect/core/dist/index.es.js:81:68100)
    at async J.start (webpack-internal:///./node_modules/@walletconnect/core/dist/index.es.js:81:67943)
    at async M.initialize (webpack-internal:///./node_modules/@walletconnect/sign-client/dist/index.es.js:37:40418)
    at async M.init (webpack-internal:///./node_modules/@walletconnect/sign-client/dist/index.es.js:37:40185)
```

```
v1.0 has been sunset. Wallets and Dapps must upgrade to v2.0 before June 28
```

It is not possible to migrate to v2 when v2 is not working.. Can you move the date at least few months further?

**Comment by francesco-gaglione** [comment]
Same error here. I cannot send the source code since it is private, but it is easy to reproduce. 
Web3Modal configuration is the standard one, from the doc page, with v2. Then serve the next.js project in dev mode and open it into the metamask browser and an error appear. 
Just a note, this error happens only in dev mode, instead all work well in production build and static serve.

**Comment by francesco-gaglione** [comment]
I was able to find the issue in the source code. I debugged the error and the error was "socket stalled" so I think it is a timeout issue since it is set to 5 seconds. 

Here the link to the source code: https://github.com/WalletConnect/walletconnect-monorepo/blob/b9aae24617a4d423c4ea3a9225176c3ffe8d13de/packages/core/src/controllers/relayer.ts#L206C43-L206C43

Is it acceptable to raise the timeout to 10 secs? If yes, I'm currently working on this, so let me know I can contribute on this

**Comment by Holybasil** [comment]
> I was able to find the issue in the source code. I debugged the error and the error was "socket stalled" so I think it is a timeout issue since it is set to 5 seconds.
> 
> Here the link to the source code: https://github.com/WalletConnect/walletconnect-monorepo/blob/b9aae24617a4d423c4ea3a9225176c3ffe8d13de/packages/core/src/controllers/relayer.ts#L206C43-L206C43
> 
> Is it acceptable to raise the timeout to 10 secs? If yes, I'm currently working on this, so let me know I can contribute on this

Looking forward to your pr 😀

**Comment by rick20220222** [comment]
I also have the same problem, how to solve this?

**Comment by francesco-gaglione** [comment]
> > I was able to find the issue in the source code. I debugged the error and the error was "socket stalled" so I think it is a timeout issue since it is set to 5 seconds.
> > Here the link to the source code: https://github.com/WalletConnect/walletconnect-monorepo/blob/b9aae24617a4d423c4ea3a9225176c3ffe8d13de/packages/core/src/controllers/relayer.ts#L206C43-L206C43
> > Is it acceptable to raise the timeout to 10 secs? If yes, I'm currently working on this, so let me know I can contribute on this
> 
> Looking forward to your pr grinning

Any contribution guide or how to link this issue to the pr? On which branch?

**Comment by andreanistico** [comment]
The mentioned merge doesn't solve the problem. V2 is still not usable and today V1 will be shutdown. Please consider to make V1 continue to work until V2 is fully working. 

Also consider that due to the removal of in-app browsers on iOS for the wallet apps, WalletConnect remains the only way to use Dapps on mobile on iOS. Don't shutdown V1.

**Comment by francesco-gaglione** [comment]
news on this? I see these merged issues. Were they published? In which libraries and versions? @bkrem

**Comment by didostap** [comment]
Same problem here
Has been any fix provided?

**Comment by eddielthomas** [comment]
I am so frustrated with this issue... I have to show my clients a working MVP in 3 hours and I am afraid of the results.   I am considering going back to Blocknative Onboarding...  I have to start to code it now to meet the dead line

**Comment by NBMSacha** [comment]
Any news on this? Impacting at least 100-200 users on our prod (mobula.fi)

**Comment by prnthh** [comment]
> I was able to find the issue in the source code. I debugged the error and the error was "socket stalled" so I think it is a timeout issue since it is set to 5 seconds.

This is probably the cause when seeing this out in production. I saw a spike in this error and I realised it was due to very poor network conditions. Worked fine on the same network after I fixed my ISP throttle.

- increase the timeout in case it eventually resolves on a slower connection
- handle the socket timeout case gracefully

**Comment by eddielthomas** [comment]
How did you fix it?  I'm actively dealing with this problem....

**Comment by eddielthomas** [comment]
How did you fix it?  I'm actively dealing with this problem....

It also doesn't work on mobile!

**Comment by kdenhartog** [comment]
This can be closed now that #3230 has been merged
